workflows/gpu_operator_dashboard/fetch_ci_data.py

```
@@ -452,69 +471,175 @@ def process_closed_prs(results_by_ocp: Dict[str, List[Dict[str, Any]]]) -> None:
+    # Convert back to sorted list for JSON serialization
+    merged_version_data["job_history_links"] = sorted(list(all_job_history_links))
+
+    return merged_version_data
+
+
 def merge_and_save_results(
-    new_results: Dict[str, List[Dict[str, Any]]],
+    new_results: Dict[str, Dict[str, List[Dict[str, Any]]]],
     output_file: str,
-    existing_results: Dict[str, Dict[str, Any]] = None
+    existing_results: Dict[str, Dict[str, Any]] = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)